src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
 
+const MAX_HEALTH = 100;
+const BITE_DAMAGE = 50;
+const DEAD_VALUE = 0;
+
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name) {
+    this.name = name;
+    this.health = MAX_HEALTH;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (
+      animal instanceof Herbivore &&
+      !(animal instanceof Carnivore) &&
+      !animal.hidden
+    ) {
+      animal.health -= BITE_DAMAGE;
+      Animal.alive = Animal.alive.filter((item) => item.health > DEAD_VALUE);
```

Comment:
The filter operation for removing dead animals from the `Animal.alive` array is correctly placed to avoid nested loops. However, ensure that there is logic to handle the case where an animal's health becomes exactly 0, as they should be considered dead and removed from the array.